pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"fmt"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	route "github.com/envoyproxy/go-control-plane/envoy/api/v2/route"
+	envoy_jwt "github.com/envoyproxy/go-control-plane/envoy/config/filter/http/jwt_authn/v2alpha"
+	http_conn "github.com/envoyproxy/go-control-plane/envoy/config/filter/network/http_connection_manager/v2"
+	"github.com/golang/protobuf/ptypes/empty"
+	"istio.io/api/security/v1beta1"
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pilot/pkg/networking/plugin"
+	"istio.io/istio/pilot/pkg/networking/util"
+	"istio.io/istio/pilot/pkg/security/authn"
+	authn_model "istio.io/istio/pilot/pkg/security/model"
+	"istio.io/pkg/log"
+)
+
+// Implemenation of authn.PolicyApplier with v1beta1 API.
+type v1beta1PolicyApplier struct {
+	jwtPolicies []*model.Config
+	// TODO: add mTLS configs.
+	// TODO: add v1alpha1 fallback configs.
+}
+
+func (a v1beta1PolicyApplier) JwtFilter(isXDSMarshalingToAnyEnabled bool) *http_conn.HttpFilter {
+	var filterConfigProto *envoy_jwt.JwtAuthentication
+	filterConfigProto = convertToEnvoyJwtConfig(a.aggregateJWTRules())
```

Comment:
nit/perf: If we call this more than once per `v1beta1PolicyApplier` we may want to precompute

Reply:
like caching into push context?

Reply:
Cache the aggregated JWTRules list.